spaceship/lib/spaceship/tunes/availability.rb

```diff
@@ -26,12 +32,33 @@ def self.from_territories(territories = [], params = {})
             territories
           end
         obj.include_future_territories = params.fetch(:include_future_territories, true)
+        obj.cleared_for_preorder = params.fetch(:cleared_for_preorder, false)
+        obj.app_available_date = params.fetch(:app_available_date, nil)
         return obj
       end
 
       def territories
         @territories ||= raw_data['countries'].map { |info| Territory.new(info) }
       end
+
+      def cleared_for_preorder
+        return @cleared_for_preorder unless @cleared_for_preorder.nil?
+
+        value = false
+        if (pre_order = raw_data['preOrder']) && (hash = pre_order['clearedForPreOrder'])
+          value = hash['value']
```

Comment:
Could this `value` lookup leverage the `attr_mapping` ability to drill down to the value as shown here? https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/tunes/app_details.rb#L47